### PR TITLE
perf: Avoid Regex for indentation when possible

### LINKED ISSUE #9715: IndentedStringBuilder uses Regex to perform indentation
### Current behavior

This method is used by `IndentedStringBuilder` and causes a lot of string allocations and performance impact by the use of Regex:
https://github.com/unoplatform/uno/blob/b671f1de73245cf57265af112bd5b3c9da4e403f/src/Uno.Foundation/Uno.Core.Extensions/Uno.Core.Extensions/StringExtensions.cs#L242

The idea behind this feature is that if there are multiple lines, all these lines should be indented the same, but in most of the generator's use of `IndentedStringBuilder`  single line strings are created. We should be changing this code to either detect if there are actually new lines in the string to avoid the use of Regex, or simply make a second version of index called `IndentMultiLine` or similar which allows for single-line indentation to be used by `IndentedStringBuilder`.

### Anything else we need to know?

Related to https://github.com/unoplatform/uno/issues/7506

### PR DESCRIPTION
**Pull request by Youssef1313**
GitHub Issue (If applicable): closes #9715

## PR Type

What kind of change does this PR introduce?

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Validated PR `Screenshots Compare Test Run` results.
- [ ] Contains **NO** breaking changes
- [ ] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [ ] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

## Other information

Internal Issue (If applicable):

### CODE REVIEW

File: src/SourceGenerators/Uno.UI.SourceGenerators/DependencyObject/DependencyObjectGenerator.cs [hunk 1]
```diff
@@ -247,30 +247,30 @@ private void WriteiOSMoveToSuperView(INamedTypeSymbol typeSymbol, IndentedString
 
 				if (overridesWillMoveToSuperview)
 				{
-					builder.AppendFormatInvariant(@"
+					builder.AppendIndented(@"
```
Comment (jeromelaban) [inline src/SourceGenerators/Uno.UI.SourceGenerators/DependencyObject/DependencyObjectGenerator.cs:250]: This is a multiline block. I don't think we should be having a "multiline" version of `AppendIndented`, but rather search for the first newline inside the block then switch to the multiline version if there is one. It'll make authoring a lot easier, even if at the expense of a lookup when adding the line.

Comment (Youssef1313) [reply src/SourceGenerators/Uno.UI.SourceGenerators/DependencyObject/DependencyObjectGenerator.cs:250]: @jeromelaban I wanted to keep `AppendIndented` as fast as possible. That's why I introduced a specific API for multiline.

Regarding authoring, actually most multiline appends are *already* broken (with and without this PR). The passed multi-line verbatim string is already indented in most cases, and we add extra indentation to that. So the authoring was and still an issue. Let me know what do you think.

File: src/Uno.Foundation/Uno.Core.Extensions/Uno.Core.Extensions.Compatibility/Extensions/IndentedStringBuilder.cs [hunk 2]
```diff
@@ -39,54 +39,62 @@ public IndentedStringBuilder(StringBuilder stringBuilder)
...
-		public virtual void AppendFormat(IFormatProvider formatProvider, string pattern, params object[] replacements)
+		public void AppendIndented(string text)
 		{
-			_stringBuilder.AppendFormat(formatProvider, pattern.Indent(CurrentLevel), replacements);
+			AppendIndented(text, CurrentLevel);
+		}
+
+		private void AppendIndented(string text, int indentCount)
+		{
+			_stringBuilder.Append(new string('\t', indentCount)); // TODO: Which is faster? this? or looping and appending a single tab each iteration?
```
Comment (ebariche) [inline src/Uno.Foundation/Uno.Core.Extensions/Uno.Core.Extensions.Compatibility/Extensions/IndentedStringBuilder.cs:84]: We should use this : https://docs.microsoft.com/en-us/dotnet/api/system.text.stringbuilder.append?view=net-6.0#system-text-stringbuilder-append(system-char-system-int32)

File: src/Uno.Foundation/Uno.Core.Extensions/Uno.Core.Extensions.Compatibility/Extensions/IndentedStringBuilder.cs [hunk 3]
```diff
@@ -39,54 +39,62 @@ public IndentedStringBuilder(StringBuilder stringBuilder)
...
+			_stringBuilder.Append(text);
+		}
+
+		public void AppendFormatIndented(IFormatProvider formatProvider, string text, params object[] replacements)
+		{
+			_stringBuilder.Append(new string('\t', CurrentLevel)); // TODO: Which is faster? this? or looping and appending a single tab each iteration?
```
Comment (ebariche) [inline src/Uno.Foundation/Uno.Core.Extensions/Uno.Core.Extensions.Compatibility/Extensions/IndentedStringBuilder.cs:90]: Same here

Comment (Youssef1313) [reply src/Uno.Foundation/Uno.Core.Extensions/Uno.Core.Extensions.Compatibility/Extensions/IndentedStringBuilder.cs]: Thanks @ebariche. Fixed

File: src/SourceGenerators/Uno.UI.SourceGenerators/DependencyObject/DependencyObjectGenerator.cs (hunk 1 above)
Comment (jeromelaban) [reply src/SourceGenerators/Uno.UI.SourceGenerators/DependencyObject/DependencyObjectGenerator.cs]: Makes sense, let's keep as is for now then. Thanks!

**Review by jeromelaban — APPROVED** [review]

**Pull request merged**